buildscripts/common.props

```diff
@@ -1,21 +1,35 @@
 <Project>
-  <PropertyGroup>
-    <VersionPrefix>5.0.0</VersionPrefix>
-    <NoWarn>$(NoWarn);CS1591</NoWarn>
-    <RepositoryType>git</RepositoryType>
-    <RepositoryUrl>https://github.com/castleproject/Core</RepositoryUrl>
-  </PropertyGroup>
+	<PropertyGroup>
+		<NoWarn>$(NoWarn);CS1591;CS3014;CS3003;CS3001;CS3021</NoWarn>
+		<RepositoryType>git</RepositoryType>
+		<RepositoryUrl>https://github.com/castleproject/Core</RepositoryUrl>
+		<BuildVersion Condition="'$(APPVEYOR_BUILD_VERSION)'==''">4.0.0</BuildVersion>
+		<BuildVersion Condition="'$(APPVEYOR_BUILD_VERSION)'!=''">$(APPVEYOR_BUILD_VERSION)</BuildVersion>
```

Comment:
>I'd prefer we used something more generic, than had the top level script populate BUILD_VERSION or something so we can do the same thing on Travis.

I need a bit more info here. Travis injects build numbers using the `TRAVIS_BUILD_NUMBER` environment variable whilst AppVeyor uses `APPVEYOR_BUILD_VERSION` for the full number or `APPVEYOR_BUILD_NUMBER`. What did you mean when you said `something more generic`? Version.props?

As for zeroing out the build version numbers, no problem! The only problem is that in some of the tests there are currently assertions using v1.0.0.0 in string asserts. But I am sure we can live with this for now. Will follow up with another commit to fix this.